update/Pos/Update/Poll/Types.hs

```diff
@@ -211,6 +213,9 @@ data PollModifier = PollModifier
     , pmActiveProps       :: !(MapModifier UpId ProposalState)
     , pmDelActivePropsIdx :: !(HashMap ApplicationName (HashSet UpId))
     , pmSlottingData      :: !(Maybe SlottingData)
+    , pmEpochProposers    :: !(Maybe (HashSet StakeholderId))
+    -- ^ Cache of DB state.
```

Comment:
>  Do you propose to split them on two different datas in this PR?

No, I propose to do it later, but in this PR I propose not to mix two different things and make `getEpochProposers` read-only. Currently it can modifiy `PollModifier` which is confusing and is bad for the reasons described above.

AFAIU, It serves only as an optimization, but I think it's negligible given that everything else is not optimized this way, so better approach is to optimize it all at once.